src/lib.rs

```diff
@@ -61,37 +64,77 @@ pub mod pallet {
     #[pallet::storage]
     pub type VMStorage<T> = StorageMap<_, Blake2_128Concat, Vec<u8>, Vec<u8>>;
 
+    /// Storage for multi-signature/signer requests.
+    #[pallet::storage]
+    pub type Multisigs<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        CallHash,
+        Multisig<BlockNumberFor<T>, T::AccountId, T::MaxScriptSigners>,
+    >;
+
     /// MoveVM pallet configuration trait
     #[pallet::config]
     pub trait Config: frame_system::Config {
+        /// The currency mechanism.
+        type Currency: Currency<Self::AccountId>
+            + ReservableCurrency<Self::AccountId>
+            + LockableCurrency<Self::AccountId>;
+
+        /// Maximum life time for requests.
+        #[pallet::constant]
+        type MaxRequestLifetime: Get<u32>;
+
+        /// Maximum number of signatories in multi-signer requests.
+        #[pallet::constant]
+        type MaxScriptSigners: Get<u32>;
+
+        /// The overarching call type.
+        type RuntimeCall: Parameter
+            + Dispatchable<RuntimeOrigin = Self::RuntimeOrigin, PostInfo = PostDispatchInfo>
+            + GetDispatchInfo
+            + From<frame_system::Call<Self>>;
+
         /// Because this pallet emits events, it depends on the runtime's definition of an event.
         type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
         /// Type representing the weight of this pallet
         type WeightInfo: WeightInfo;
-
-        /// The currency mechanism.
-        type Currency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
     }
 
     // Pallets use events to inform users when important changes are made.
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
+        /// Event about successful move-bundle published.
+        /// [account]
+        BundlePublished { who: T::AccountId },
         /// Event about calling execute function.
         /// [account]
         ExecuteCalled { who: T::AccountId },
-
-        /// Event about successful move-module publishing
+        /// Event about a script execution with multiple signers.
+        /// [vec<account>]
+        ExecutionAsMulti {
+            who: BoundedVec<T::AccountId, T::MaxScriptSigners>,
+        },
```

Comment:
Can be removed

Reply:
Maybe we can change the original event to have a vector of signers
```
ExecuteCalled { who: T::AccountId },
```

Reply:
Done :+1: